From: "Zenghui Yu (Huawei)" <zenghui.yu@linux.dev>
Date: Mon, 31 Aug 2026 07:26:03 -0700
Subject: [PATCH] mm/damon/core: remove declaration of __damon_commit_ctx()

Patch series "mm/damon: misc cleanups".

Cleanup the code, tests and samples for clarifications and readability.

The patches are individually sent by the authors.  I'm reposting those as
one series for convenience of handling.  For this reason, changelog is on
each patch's commentary section.


This patch (of 7):

__damon_commit_ctx() was added by commit b1471afe4d10 ("mm/damon/core: do
parameter testing commit on damon_start()") but is actually not needed.
Remove it.

Link: https://lore.kernel.org/20260831142611.77572-1-sj@kernel.org
Link: https://lore.kernel.org/20260831142611.77572-2-sj@kernel.org
Signed-off-by: Zenghui Yu (Huawei) <zenghui.yu@linux.dev>
Signed-off-by: SJ Park <sj@kernel.org>
Reviewed-by: SJ Park <sj@kernel.org>
Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Cc: Shuah Khan <shuah@kernel.org>
Cc: Zenghui Yu <zenghui.yu@linux.dev>
Cc: Enze Li <lienze@kylinos.cn>
Cc: Hari Mishal <harimishal1@gmail.com>
Cc: Jaeyeon Lee <jaeyeon.lee.dev@gmail.com>
Cc: Li Youhong <liyouhong@kylinos.cn>
Cc: zhaozhengzhuo <zhaozhengzhuo@uniontech.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/damon/core.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 3632b06f1db58..718195fbeb686 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1935,8 +1935,6 @@ static int __damon_start(struct damon_ctx *ctx)
 	return err;
 }
 
-static int __damon_commit_ctx(struct damon_ctx *dst, struct damon_ctx *src);
-
 /**
  * damon_start() - Starts the monitorings for a given group of contexts.
  * @ctxs:	an array of the pointers for contexts to start monitoring
-- 
2.47.3

